src/shared/lib/styledComponents/styled.d.ts

```diff
@@ -0,0 +1,8 @@
+import "styled-components";
+import { ColorsTypes, FontsTypes } from "./theme";
+declare module "styled-components" {
+  export interface DefaultTheme {
+    colors: ColorsTypes;
+    fonts: FontsTypes;
+  }
+}
```

Comment:
P3. 이렇게 작성해도 기능에 문제 없을 것 같은데 어때?

```suggestion
import { ColorsTypes, FontsTypes } from "./theme";
declare module "styled-components" {
 interface DefaultTheme {
    colors: ColorsTypes;
    fonts: FontsTypes;
  }
}
```

Reply:
https://styled-components.com/docs/api#typescript

이부분은 공식문서에 예시를 참고한건데 준근이 말대로 없어도 동작이 동작에는 문제가 없는데 공식문서에서 
import "styled-components" 부분은 original module 선언을 import하고 그 아래에서 확장한다고 나와있어서 적용한거거든 
내가 적용한 방식이 값을 먼저 지정하고 거기서 타입을 빼오는 방식이라 필요없는건가??

Reply:
흐음 나도 이유를 잘 모르겠네.. 문서가 오래된 버전인 걸지도? 우리가 설치한 버전 기준으로 DefaultTheme의 JSDoc 주석에는 이렇게 작성되어 있어.

```
/**
 * Override DefaultTheme to get accurate typings for your project.
 *
 * ```
 * // create styled-components.d.ts in your project source
 * // if it isn't being picked up, check tsconfig compilerOptions.types
 * import type { CSSProp } from "styled-components";
 * import Theme from './theme';
 *
 * type ThemeType = typeof Theme;
 *
 * declare module "styled-components" {
 *  export interface DefaultTheme extends ThemeType {}
 * }
 *
 * declare module "react" {
 *  interface DOMAttributes<T> {
 *    css?: CSSProp;
 *  }
 * }
 * ```
 */
 ```